pkg/resources/grant_helpers.go

```diff
@@ -54,12 +55,21 @@ type grant struct {
 // splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
 // returns the object name and privilege.
 func splitGrantID(v string) (string, string, string, string, error) {
-	arr := strings.Split(v, "|")
-	if len(arr) != 4 {
-		return "", "", "", "", fmt.Errorf("ID %v is invalid", v)
+
+	fmt.Printf("[DEBUG] splitGrantID input: %s", v)
+	reader := csv.NewReader(strings.NewReader(v))
+	reader.Comma = '|'
+
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return "", "", "", "", err
+	}
+
+	if (len(lines) == 1) && (len(lines[0]) == 4) {
```

Comment:
I would do this the other way around, have the failure error up front and error out early if so.

```
if (len(lines) != 1 || len(lines[0]) !=4) {
    return """""', err
}
```

this reduces the cognitive load on the readers of the function since the further you go the less conditions you have to remember